pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -353,26 +352,33 @@ func getMutualTLSMode(mtls *v1beta1.PeerAuthentication_MutualTLS) model.MutualTL
 // replaced with config from workload-level, UNSET in workload-level config will be replaced with
 // one in namespace-level and so on.
 func composePeerAuthentication(rootNamespace string, configs []*model.Config) *v1beta1.PeerAuthentication {
-	var meshPolicy, namespacePolicy, workloadPolicy *v1beta1.PeerAuthentication
-	// Creation time associate with the selected workloadPolicy above. Initially set to max time.
-	workloadPolicyCreationTime := time.Unix(1<<63-1, 0)
+	var meshCfg, namespaceCfg, workloadCfg *model.Config
 
 	for _, cfg := range configs {
 		spec := cfg.Spec.(*v1beta1.PeerAuthentication)
-		if cfg.Namespace == rootNamespace && spec.Selector == nil {
-			meshPolicy = spec
-		} else if spec.Selector == nil {
-			namespacePolicy = spec
+		if spec.Selector == nil || len(spec.Selector.MatchLabels) == 0 {
```

Comment:
This config hierachy case switch has occurred several times in previous PR also. Having a utility func accept config and return the type (root/ns/wl) might be clearer. (optional)

Reply:
I would suggest you really look at the sidecar API that resolves all of this way before the XDS request processing. Not really a fan of timestamp.before and after. A simple sort and picking the first non label policy, followed by breaking out of the for loop is easier. IMO, the more this logic diverges from how the sidecar API is done in push context, the more bugs you will hit as you hit corner cases in future.